variants/DISCO_L072CZ_LRWAN1/variant.h

```diff
@@ -0,0 +1,162 @@
+  See the GNU Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public
+  License along with this library; if not, write to the Free Software
+  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+*/
+
+#ifndef _VARIANT_ARDUINO_STM32_
+#define _VARIANT_ARDUINO_STM32_
+
+/*----------------------------------------------------------------------------
+ *        Headers
+ *----------------------------------------------------------------------------*/
+
+#include "pins_arduino.h"
+
+#ifdef __cplusplus
+extern "C"{
+#endif // __cplusplus
+
+/*----------------------------------------------------------------------------
+ *        Pins
+ *----------------------------------------------------------------------------*/
+extern const PinName digitalPin[];
+
+enum {
+  PA3,  //D0
+  PA2,  //D1
+  PA10, //D2
+  PB13, //D3
+  PB5,  //D4
+  PB7,  //D5
+  PB2,  //D6
+  PA8,  //D7
+  PA9,  //D8
+  PB12, //D9
+  PB6,  //D10
+  PB15, //D11
+  PB14, //D12
+  PA5,  //D13
+  PB9,  //D14
+  PB8,  //D15
+// Connector CN2
+  NC_1, //D16 - BOOT0
+  PA13, //D17 - SWD
+  PA14, //D18 - SWD
+  PH1,  //D19
+  PH0,  //D20
+// Connector CN3
+  PA1,  //D21
+  PC2,  //D22
+  PC1,  //D23
+  PA12, //D24
+  PA11, //D25
+  PA0,  //D26/A0
+  PA0_2,//D27/A1 - alias for A0 - requires closing solder bridge SB7
+  PA4,  //D28/A2
+  PA4_2,//D29/A3 - alias for A2 - requires closing solder bridge SB8
+  PB9_2,//D30/A4 - requires closing solder bridge SB11
+  PB8_2,//D31/A5 - requires closing solder bridge SB12
+  // Duplicated pins in order to be aligned with PinMap_ADC
+  PA5_2,//D32/A6  = D26
+  PEND
+};
+
+enum {
+  A_START_AFTER = D25,
+  A0,  A1,  A2,  A3,  A4,  A5,  A6,
+  AEND
+};
+
+//ADC resolution is 12bits
+#define ADC_RESOLUTION          12
+#define DACC_RESOLUTION         12
+
+//PWR resolution
+#define PWM_RESOLUTION          8
+#define PWM_FREQUENCY           1000
+#define PWM_MAX_DUTY_CYCLE      255
+
+//On-board LED pin number
+#define LED_BUILTIN             13
+#define LED_LD2                 LED_BUILTIN
+#define LED_GREEN2              LED_BUILTIN
+#define LED_GREEN               4
+#define LED_LD1                 LED_GREEN
+#define LED_BLUE                10
+#define LED_LD3                 LED_BLUE
+#define LED_RED                 5
+#define LED_LD4                 LED_RED
+
+//On-board user button
+#define USER_BTN                6
+
+
+//SPI definitions
+#define SS                      10
+#define SS1                     4
+#define SS2                     7
+#define SS3                     8
+#define MOSI                    11
+#define MISO                    12
+#define SCK                     3
+
+//I2C Definitions
+#define SDA                     14
+#define SCL                     15
+
+//Timer Definitions
+//Do not use timer used by PWM pins when possible. See PinMap_PWM.
+#define TIMER_TONE              TIM22
+#define TIMER_UART_EMULATED     TIM21
```

Comment:
I removed the  TIMER_UART_EMULATED  as it need to be reviewed. (move as SoftwareSerial library)